### PR TITLE
Fix #15193: Crash when rides/stalls are demolished

### LINKED ISSUE #15193: Demolishing stalls sometimes causes the game to crash
### Operating System

Linux (x86-64)

### OpenRCT2 build

OpenRCT2, v0.3.4.1-141 (3ed2f8e98 on makepkg)

### Describe the issue

Some stalls, when demolished, cause the game to crash, most notably the Drink Stall and Information Kiosk.

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Place Drink Stall/Information Kiosk
2. Delete the kiosk
3. Game crashes

### Attachments

https://user-images.githubusercontent.com/46719718/128779462-be933c00-0dea-4340-912c-3b946527c0a0.mp4

### PR DESCRIPTION
**Pull request by ZehMatt**
It crashed because it went out of bounds on the peep thoughts. I cleaned up the logic and also moved the function into the Guest class.

### CODE REVIEW

File: src/openrct2/peep/Guest.cpp [hunk 1]
```diff
@@ -7356,3 +7356,115 @@ bool Guest::HasItem(ShopItem peepItem) const
...
+    return false;
+}
+
+void Guest::RemoveRideFromMemory(ride_id_t rideId)
+{
+    uint8_t ride_id_bit = rideId % 8;
+    uint8_t ride_id_offset = rideId / 8;
+
+    // clear ride from potentially being in RidesBeenOn
+    RidesBeenOn[ride_id_offset] &= ~(1 << ride_id_bit);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp:7380]: Access to RidesBeenOn should be in a method of its own, I’d say, as accessing bits in it is not trivial. It also makes it easier to refactor (there is a refactor from Duncan coming up).

Comment (ZehMatt) [reply src/openrct2/peep/Guest.cpp:7380]: Yeah I spoke to Duncan, this will be replaced with his methods later.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Freezes the game.

To reproduce:
1. Open Blackpool Pleasure Beach
2. Open the thoughts of Aaron F. (the first guest in the list)
3. Demolish Turtle Chase

**Comment by ZehMatt** [comment]
> Freezes the game.
> 
> To reproduce:
> 
> 1. Open Blackpool Pleasure Beach
> 2. Open the thoughts of Aaron F. (the first guest in the list)
> 3. Demolish Turtle Chase

Should be good now.

**Review by Gymnasiast — APPROVED** [review]
Seems to work correctly now. I think it still needs a changelog update.

**Pull request merged**